tests/test_main.py

```diff
@@ -102,6 +102,45 @@ def test_run_sql(dsn, log_output):
     ]
 
 
+@pytest.mark.parametrize(
+    "sql_query",
+    [
+        "CREATE TABLE #test(patient_id int); INSERT #test VALUES (1); SELECT patient_id from #test;",
+        "SELECT 1 as patient_id",
+    ],
+)
+def test_run_sql_include_statistics(dsn, log_output, sql_query):
+    results = main.run_sql(dsn=dsn, sql_query=sql_query, include_statistics=True)
+    assert list(results) == [{"patient_id": 1}]
+
+    by_event = {e["event"]: e for e in log_output.entries}
```

Comment:
`by_event` is a `dict`. Essentially, we've grouped `log_output.entries` by the value of each entry's `event`. This means that we don't have to write a test that relies on the order of events in the log, which makes this version of the test less brittle than the last version of the test. (We may wish to improve our approach to grouping, should we log the same type of event more than once.)

Reply:
Yeah, it was just the name "by_event" didn't immediately make sense to me - no biggie at all!